### PR TITLE
Update TDF spec to latest design

### PR DESCRIPTION
**Pull request by bleggett**
### Proposed Changes

* `ClaimsAttributeObject` and `AttributeObject` are the same thing - so deleted the former and clarified that
  * If an `AttributeObject` is used in a PolicyObject, it is a **object attribute** 
  * If an `AttributeObject` is used in a ClaimsObject, it is a **subject attribute**
  * ...and generally get us closer to the [NIST terminology](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-162.pdf) (thanks Dave)
* Remove `EntityObject` from the spec - it doesn't exist anymore
* Remove schema level versions. Every schema change in this spec would necessitate a Semver version bump/tag of this repo by definition - so this repo's semver is == the TDF spec version == the version of every schema in the spec - there is no point in maintaining two versions for the same thing.
* Drop CHANGELOG - it's redundant with Git. All PRs should bump the Git tag for the repo with an annotated tag describing the change, and use semver rules for updating the tag version. You can then get a `changelog` by running `git tag -n` on the primary branch.

### Checklist

- [x] A clear description of the change has been included in this PR.
- [x] The changelog has been updated.
- [x] All schema validation tests have been updated appropriately and are passing.
- [x] Version numbers have been updated as per the [Versioning Guidelines](../CONTRIBUTING.md#verison-changes).
- [x] Major/minor version changes only: A writeup has been included discussing the motivation and impact of this change.
- [x] Major/minor version changes only: The minimum wait time has elapsed.
- [x] Project version changes only: I have updated both the `VERSION` file and the `git tag`.
- [x] This change otherwise adheres to the project [Contribution Guidelines](../CONTRIBUTING.md).
- [x] Every item in this checklist has been completed.

### CODE REVIEW

**Review by JeffGradyAtVirtru — DISMISSED** [review]
LGTM.  Two very minor typos that may have existed prior.

File: protocol/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The canonical architecture contains four major components.
     * Return the resulting Claims Object in the signed IdP JWT.
   * A list of Certified OpenID Connect applications can be found at:  https://openid.net/developers/certified/
   * *Virtru Protocol Mapper* is our Keycloak-specific reference implementation of the above functionality.
-* *Attribute Provider* (AP) - A web service that will receive requests from our OIDC IdP custom code (ex: Virtru Protocol Mapper) containing information about authenticated subjet and return our custom OIDC/OAuth claims in response.  It is the responsibility of Attribute Provider to transform incoming 3rd party IdP claims/metadata to outgoing ABAC policy attributes.  It returns a TDF [Claims Object](../schema/ClaimsObject.md).
+* *Attribute Provider* (AP) - A web service that will receive requests from our OIDC IdP custom code (ex: Virtru Protocol Mapper) containing information about authenticated subjet and return our custom OIDC/OAuth claims in response.  It is the responsibility of Attribute Provider to transform incoming 3rd party IdP claims/metadata to a set of outgoing [Attribute Objects](../schema/AttributeObject.md).  It returns a TDF [Claims Object](../schema/ClaimsObject.md).
```
Comment (dmihalcik-virtru) [inline protocol/README.md:18]: I am fine with introverted attributes.

File: schema/AttributeObject.md [hunk 2]
```diff
@@ -22,15 +32,15 @@ The current schema version is `1.1.0`.
   "displayName": "classification",
   "pubKey": "pem encoded public key of the attribute",
   "kasUrl": "https://kas.example.com/",
-  "schemaVersion:": "x.y.z"
+  "tdf_spec_version:": "x.y.z"
```
[the remark this answers is not in the document]
Comment (dmihalcik-virtru) [reply schema/AttributeObject.md:35]:
```suggestion
  "tdfVersion:": "x.y.z"
```
maybe?

**Review by dmihalcik-virtru — DISMISSED** [review]

File: protocol/README.md (hunk 1 above)
Comment (bleggett) [reply protocol/README.md:18]: 👍 They're outgoing but introverted, as opposed to outgoing but extroverted.

**Review by dmihalcik-virtru — DISMISSED** [review]

**Comment by dmihalcik-virtru** [comment]
Don't forget to squash and merge.

I'd kinda you refactor these into focused PRs for the following, but I'd rather this get in than not so it is up to you.

1. Switch away from explicit CHANGE LOG to conventional commits 
2+. OpenAPI cleanups, possibly split by theme

See: https://www.theregister.com/2021/09/06/github_merges_useless_garbage_says/

**Comment by bleggett** [comment]
@dmihalcik-virtru He's right that WebUI commits are kinda gross, but I don't know anyone that uses those.

As to the rest, you either get the complete history or you squash it and lose info - there's no in-between.

But he knows that, he wrote Git. It's good to know he's frustrated at his own creation though, humanizes him a bit.

**Comment by bleggett** [comment]
@biscoe916 @sujankota It would be good to get your eyes/approvals/objections on this.

**Review by dmihalcik-virtru — DISMISSED** [review]

File: schema/ClaimsObject.md [hunk 3]
```diff
@@ -2,66 +2,58 @@
 
 ## What is this?
 
-The Claims Object is a JSON object used to ensure that TDF Clients
+The Claims Object is a JSON object used to ensure that actors
```
[the remark this answers is not in the document]
Comment (biscoe916) [reply schema/ClaimsObject.md:5]: I also prefer `subject`.

File: protocol/README.md (hunk 1 above)
Comment (biscoe916) [reply protocol/README.md:18]: Nit: There are a couple grammar/spelling mistakes here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -7,26 +7,27 @@ Contributions to the TDF3 specification are welcome! Please be sure to follow th
...
-* **Schema version** - Shared data objects are versioned indepedent of the project version. If the schema for any of these objects changes, then the corresponding schema version should be changed accordingly.
-* **API version** - Each swagger-defined API has an associated version number. If the API changes then the corresponding swagger version number should be changed accordingly.
+The spec version is this repo's most recent semver Git tag - this means that if the spec version is 4.1.0, then the protocol version and all associated schema versions are also 4.1.0
 
 Any changes that affect _project_ version must update both `git tag` and the [VERSION](VERSION) file.
 
-Any changes that affect _schema_ versions must add or update [validation tests](schema/test/) accordingly.
+Rather than use a changelog, we ask that you use annotated `git tags` when bumping the spec Semver, and use the annotation message to describe the change.
+> Example: `git tag -a 4.1.0 -m "Spec version 4.1.0 - twiddled a doohickey"`)
```
Comment (biscoe916) [inline CONTRIBUTING.md:28]: I have a slight preference for changelogs. Whats the argument against?

**Review by biscoe916 — CHANGES_REQUESTED** [review]

File: schema/ClaimsObject.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (bleggett) [reply schema/ClaimsObject.md:5]: That's a majority then - I will update to use `subject` everywhere.

File: CONTRIBUTING.md (hunk 4 above)
Comment (bleggett) [reply CONTRIBUTING.md:28]: Arguments against a CHANGELOG file

- It is "another manual artifact" we have to update, keep in sync, validate, and (usually) fix/tidy for a release
- The CHANGELOG artifact itself is by definition a duplicate of Git history, especially if we use the `git tagging` method described here
- Using Git tags means we can automatically generate a changelog that agrees with what's actually in Git, with one command, always.
- Using Git tags means we can actually leverage Github's release UX, which is what people looking at this repo will expect anyway.

**Review by sujankota — APPROVED** [review]

**Review by JeffGradyAtVirtru — APPROVED** [review]
LGTM

**Review by biscoe916 — APPROVED** [review]
👍

**Pull request merged**